src/main.py

```diff
@@ -1,6 +1,597 @@
+
+    org, app_client_id, aws_default_region, aws_secret_name = get_environment_variables()
+
+    logger.log_info("Environment variables retrieved.")
+
+    # Create Boto3 Secret Manager client
+
+    session = boto3.session.Session()
+    secret_manager = session.client(service_name="secretsmanager", region_name=aws_default_region)
+
+    logger.log_info("Boto3 Secret Manager client created.")
+
+    # Create GitHub API interfaces (GraphQL and REST)
+
+    token = get_access_token(secret_manager, aws_secret_name, org, app_client_id)
+
+    logger.log_info("Access token for GitHub API retrieved.")
+
+    ql = github_api_toolkit.github_graphql_interface(token[0])
+    rest = github_api_toolkit.github_interface(token[0])
+
+    logger.log_info("GitHub API interfaces created.")
+
+    # Get the repositories from GitHub
+
+    repositories, number_of_pages = get_repositories(logger, ql, org, archive_rules)
+
+    logger.log_info(f"Found {len(repositories)} repositories in {number_of_pages} page(s).")
+
+    # Load the archive rules from the configuration file
+
+    archive_threshold, notification_period, notification_issue_tag, exemption_filename, maximum_notifications = (
+        load_archive_rules(archive_rules)
+    )
+
+    notification_issue_title = "Repository Archive Notice"
+    notification_issue_body_tuple = (
+        "## Important Notice \n\n",
+        f"This repository has not been updated in over {archive_threshold} days and will be archived in {notification_period} days if no action is taken. \n",
+        "## Actions Required to Prevent Archive \n\n",
+        f"1. Update the repository by creating/updating a file called `{exemption_filename}`. \n",
+        "   - This file should contain the reason why the repository should not be archived. \n",
+        "   - If the file already exists, please update it with the latest information. \n",
+        "2. Close this issue. \n\n",
+        f"After these actions, the repository will be exempt from archive for another {archive_threshold} days. \n\n",
+        "If you have any questions, please contact an organization administrator.",
+    )
+
+    notification_issue_body = "".join(notification_issue_body_tuple)
+
+    # Iterate over the repositories, creating issues and archiving where necessary
+
+    interfaces = [logger, rest]
+    archive_criteria = [
+        str(archive_threshold),
+        str(notification_period),
+        str(notification_issue_tag),
+        str(maximum_notifications),
+    ]
+    notification_content = [notification_issue_title, notification_issue_body]
+
+    repositories_archived, issues_created = process_repositories(
+        interfaces, org, repositories, archive_criteria, notification_content
+    )
+
+    message = f"Script completed. {len(repositories)} repositories checked. {issues_created} issues created. {repositories_archived} repositories archived."
+
+    logger.log_info(message)
+
+    return message
+
+
+# if __name__ == "__main__":
```

Comment:
I'd probably err on just mentioning this in the readme or have a comment to say it is for dev purposes only